### PR TITLE
Issue-5: Adds tx-time support

### PR DESCRIPTION
**Pull request by naomijub**
Solves https://github.com/naomijub/gXTDB/issues/5

### CODE REVIEW

File: test/gxtdb/logic/time_test.clj [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+(def base-vector [:a :b])
+
+(deftest conj-time-ranges-test
+  (testing "When init and end rages are nil, base vector is unchanged"
+    (is (= base-vector (conj-some-valid-timerange base-vector nil nil))))
+  (testing "When only end time is not nil, base vector is unchanged"
+    (is (= base-vector (conj-some-valid-timerange base-vector nil :d))))
+  (testing "When only init time is not nil, base vector is appended of init-time"
+    (is (= [:a :b :c] (conj-some-valid-timerange base-vector :c nil))))
+  (testing "When only init time and end itme are not nil, base vector is appended of init-time and end-time"
```
Comment (jerielverissimo) [inline test/gxtdb/logic/time_test.clj:15]:
```suggestion
  (testing "When only init time and end time are not nil, base vector is appended of init-time and end-time"
```

File: test/gxtdb/service_test.clj [hunk 2]
```diff
@@ -47,21 +47,33 @@
...
-  (testing "Submit a put tx to xtdb-node"
-    (let [tx @(client/SubmitTx
-               @(connect {:uri (str "http://localhost:" (:port @test-env))})
-               {:tx-ops [{:transaction-type
-                          {:put {:id-type :keyword, :xt-id "id1", :document {:fields {"key" {:kind {:string-value "value"}}}}}}}]})]
-      (is (inst? (-> tx :tx-time utils/->inst)))
-      (is (>=
-           (:tx-id tx)
-           0))))
+  #_(testing "Submit a put tx to xtdb-node"
```
Comment (jerielverissimo) [inline test/gxtdb/service_test.clj:50]: this test is supposed to be commented on?

**Review by jerielverissimo — APPROVED** [review]

File: Makefile [hunk 3]
```diff
@@ -1,21 +1,40 @@
...
+
+lint: clippy kibit
+
+fmt: cljfmt rsfmt
+
+fix: fmt lint
+
+fix-clj: cljfmt kibit
+
+fix-rs: rsfmt clippy
```
Comment (pfeodrippe) [inline Makefile:21]: 💯

File: test/gxtdb/service_test.clj [hunk 4]
```diff
@@ -47,21 +47,33 @@
...
+        (is (>=
+             (:tx-id tx)
+             0))))
 
   (testing "Submit a match tx to xtdb-node"
     (let [tx @(client/SubmitTx
                @(connect {:uri (str "http://localhost:" (:port @test-env))})
                {:tx-ops [{:transaction-type
-                          {:match {:id-type :keyword, :valid-time "123" , :document-id "id1", :document {:fields {"key" {:kind {:string-value "value"}}}}}}}]})]
+                          {:match {:id-type :keyword :document-id "id1" :document {:fields {"key" {:kind {:string-value "value"}}}} :valid-time  {:value {:some "2023-06-12T21:32:44.717-05:00"}}}}}
```
Comment (pfeodrippe) [inline test/gxtdb/service_test.clj:74]: How does `:valid-time` interfere with this test?

File: src/gxtdb/adapters/tx_time.clj [hunk 5]
```diff
@@ -0,0 +1,5 @@
+(ns gxtdb.adapters.tx-time
+  (:require [gxtdb.utils :as utils]))
+
+(defn ->clj-time [str-time]
+  (some-> str-time :value :some utils/->inst))
```
Comment (pfeodrippe) [inline src/gxtdb/adapters/tx_time.clj:5]: `str-time` may lead us to believe that this is a string, should it be called something like `optional-time`? 

Also, this is assuming that the time will be optional, correct? But we can change later if we see that some API requires a tie param o/

**Review by pfeodrippe — COMMENTED** [review]
LGTM, just some comments o/

File: test/gxtdb/service_test.clj (hunk 2 above)
Comment (naomijub) [reply test/gxtdb/service_test.clj:50]: Nope

File: test/gxtdb/service_test.clj (hunk 4 above)
Comment (naomijub) [reply test/gxtdb/service_test.clj:74]: it is an optional field that changes the time a match is supposed to happen/validate. Also, there was a missing element that I only detected with this change as it broke arity

File: src/gxtdb/adapters/tx_time.clj (hunk 5 above)
Comment (naomijub) [reply src/gxtdb/adapters/tx_time.clj:5]: Converted into issue

**Pull request merged**